Frameworks/ImageIO/CGImageSource.mm

```diff
@@ -112,6 +115,890 @@ - (CFStringRef)getImageType {
+    PROPVARIANT propertyValue;
+    PropVariantInit(&propertyValue);
+    NSMutableDictionary* properties = [[NSMutableDictionary alloc] init];
+
+    // JPEG Properties - Common
+    // DPIHeight and DPIWidth are saved in different places for different image formats, and these locations only represent
+    // DPI if {ushort=1} is 1, otherwise they are other units.
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=282}", &propertyValue)) && propertyValue.vt == VT_UI8) {
+        [properties setObject:[NSNumber numberWithDouble:(double)propertyValue.uhVal.LowPart/propertyValue.uhVal.HighPart]
+                       forKey:(id)kCGImagePropertyDPIWidth];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=283}", &propertyValue)) && propertyValue.vt == VT_UI8) {
+        [properties setObject:[NSNumber numberWithDouble:(double)propertyValue.uhVal.LowPart/propertyValue.uhVal.HighPart]
+                       forKey:(id)kCGImagePropertyDPIHeight];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=256}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyPixelWidth];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=256}", &propertyValue)) && propertyValue.vt == VT_UI4) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.ulVal] forKey:(id)kCGImagePropertyPixelWidth];
+    }
+    
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=257}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyPixelHeight];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=257}", &propertyValue)) && propertyValue.vt == VT_UI4) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.ulVal] forKey:(id)kCGImagePropertyPixelHeight];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=258}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyDepth];
+    }
+
+    PropVariantClear(&propertyValue);
+    if (SUCCEEDED(imageMetadataReader->GetMetadataByName(L"/app1/ifd/{ushort=274}", &propertyValue)) && propertyValue.vt == VT_UI2) {
+        [properties setObject:[NSNumber numberWithInt:propertyValue.uiVal] forKey:(id)kCGImagePropertyOrientation];
+    }
+
+    // JPEG Properties - Format-specific
+    NSMutableDictionary* jfifProperties = [[NSMutableDictionary alloc] init];
```

Comment:
Can you elaborate on why we can't use ARC in this file?  In the meantime, you can use StrongId<NSMutableDictionary> jfifProperties (IIRC from SmartTypes.h) in all places, which will handle calling release when the objects go out of scope.  Otherwise, you'd have to add 'release' calls everywhere you transfer ownership of the sub-property collections to the parent dictionary.

---
In reply to: [60630877](https://github.com/Microsoft/WinObjC/pull/445#discussion_r60630877) [](ancestors = 60630877)